### DISCUSSION TITLE
How to execute PQ with excel file?

### DISCUSSION THREAD

**Discussion by LeonStra** [microsoft/vscode-powerquery-sdk#292] [Q&A]
Is there a connector to open excel files in the PQ SDK or how can I execute a program like this?:

`let
    Source = Excel.Workbook(Web.Contents("Table1.xlsx"), null, true)
in
    Source`

**Comment by mattmasson** [comment]
The SDK supports the development of Custom Connectors. It's not meant for general evaluation of M queries. You could create a custom connector that retrieves an .xlsx file with a `Web.Contents` function call, and then parses it with `Excel.Workbook`. You could also use the M query you provided as a test file in your connector project, as long as you set credentials for the Web call.

**Reply by LeonStra** [reply]
>  It's not meant for general evaluation of M queries.

Oh, I thought it is. But is it still possible or is there a tool where you are able to evaluate your M queries? Because the editor in excel is very limited.

If it's still possible to to evaluate your M queries with this SDK, is there any example for a connecter?

Thanks in advance :)
